modular_robot/revolve2/modular_robot/body/base/_body.py

```diff
@@ -27,47 +27,20 @@ def grid_position(self, module: Module) -> Vector3:
 
         :param module: The module to calculate the position for.
         :returns: The calculated position.
-        :raises NotImplementedError: In case a module is encountered that is not supported.
         """
         position = Vector3()
 
         parent = module.parent
         child_index = module.parent_child_index
         while parent is not None and child_index is not None:
-            child = parent.children[child_index]
+            child = parent.children.get(child_index, None)
             assert child is not None
             assert np.isclose(child.rotation % (math.pi / 2.0), 0.0)
 
+            rotation = Quaternion.from_eulers((0.0, 0.0, math.pi / 2.0 * child_index))
```

Comment:
I do not like this. It simply happens to be that the orientation+child indices of active hinge, brick, core overlap, but future modules could use completely different angles.

Reply:
to be fair by that time a 'grid' doesn't make sense. but in general I don't want a coupling between the various modules based on the child index

Reply:
what if the rotation offset (pi/2) would be a constant in the parent module?
then this could handle any arbitrary shape of modules.

When looking at the V2 my idea is that each module has faces (i.e, the child indices now) and some modules can be placed on each face. But we would do a lot of if/else checks in that case if it was the original structure.

Reply:
I do not think we can make that assumption. in the future there might be modules on top or modules that are not square? we don't really know. I think it makes sense to note the angle/position for each module type seperately

Reply:
yes i see, maybe we can use the previous structure and associate the angles to the attachment point. This would also be good for the future attachment point type you have hinted.